x-pack/plugins/maps/public/embeddable/map_component.tsx

```diff
@@ -94,7 +95,18 @@ export class MapComponent extends Component<Props, State> {
         mapCenter: this.props.mapCenter,
       }
     );
-    this._mapEmbeddable.setOnInitialRenderComplete(this.props.onInitialRenderComplete);
+
+    if (this.props.onInitialRenderComplete) {
+      this._mapEmbeddable
+        .getOnRenderComplete$()
```

Comment:
Replaced setOnInitialRenderComplete with getOnRenderComplete$ to provide a single solution that will work when maps is embeddabed in another embeddable (lens, legacy visualization tilemap, and legacy visualization regionmap).